### ISSUE TITLE
[java] MissingOverride rule warns of method with name 'method'. No error reported.

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2007]
**Affects PMD Version:**
6.17.0

**Rule:**
MissingOverride

**Description:**

Following output running rule against code below the first time:
Buildfile: /data/data/com.termux/files/home/LearnJava/Puzzles/build.xml

pmd:
MissingOverride encountered unexpected method method
MissingOverride encountered unexpected method method
Analysis cache created

BUILD SUCCESSFUL
Total time: 57 seconds

No errors reported against this rule.

 Why is this warning displayed? 'method' is not a keyword or reserved word.
Do we need a new rule that disallows a method named 'method'? Should it be flagged? It's quite common to encounter methods with names 'methodX+'.

Code Sample demonstrating the issue:**

```java
package javapuzzles;

public enum Meno {
;

  public static void main(String... args) {
    method();
  }

  public static void method(int... a) {
    System.out.println("1");
  }
  
@SuppressWarnings("PMD.AvoidUsingShortType")
  public static void method(short... b) {
    System.out.println("2");
  }
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Shouldn't matter. Running through Ant and CLI.

**Comment by oowekyala** [comment]
Hi, as a user you don't need to care about that warning. The rule is not telling you that the method is badly named or anything. It reports this warning when hitting an implementation limitation, or when the `class` file on the classpath for the current file is out of sync with the source. I think, the warning should probably be logged with a finer log level in the future, and other mechanisms should be used to take care of reporting invalid classpath configuration.

**Comment by Fernal73** [comment]
The warning ought not to be visible to the user; it's presence seems to suggest that there is something wrong with the method name. It can be ignored only so long before the user renames the method simply to avoid seeing the warning. I'm fine with logging it at a finer level as long it does not make this not so subliminal suggestion to the user to change his ways.
